src/SecretsFile.hs

```diff
@@ -197,7 +197,7 @@ secretP version mount = do
 -- Please open a ticket if you require looser restrictions.
 secretVarP :: Parser String
 secretVarP = do
-  var <- intercalate "_" <$> sepBy1 (some MPC.alphaNumChar) (MPC.string "_")
+  var <- intercalate "_" <$> sepBy1 (some MPC.alphaNumChar) (some (MPC.string "_"))
```

Comment:
This doesn't actually bind `var` to a correct value. It will parse `vars__with__double__underscores` but that example will just be set in the environment as `vars_with_double_underscores`

```haskell
sepBy1 (some MPC.alphaNumChar) (some (MPC.string "_")) :: Parser [String]
```

This turn `vars__with__double__underscores` into `["vars", "with", "double", "underscores"]` (in a `Parser` context)

```haskell
(intercalate "_" ["vars", "with", "double", "underscores"]) == "vars_with_double_underscores"
```

Reply:
Another issue with this parser is that it doesn't accept environment variables with leading underscores, (which is apparently also allowed). It also accepts leading numbers (which isn't allowed)